posts/app.py

```diff
@@ -53,6 +53,49 @@ def get_all_posts_for_event(event_id):
     """Get all posts matching the event with the specified ID."""
 
 
+def upload_new_post_to_db(post, collection):
+    """Uploads a new post to the db collection.
+
+    Assumes the event matching the post's `event_id` and the user matching
+        the post's `author_id` both exist. I.e. Caller should check this.
+
+    Args:
+        post (dict): Post to add. Requires exactly the following attributes:
+            event_id (str): id of the event to post to
+            author_id (str): user id of the user making the post
+            text (str): text description
+            files (list): list of string encoded files
+        collection: pymongo collection to insert into.
+
+    Returns:
+        ObjectID: DB ID of the post that was uploaded.
+
+    Raises:
+        ValueError: Has no text body (i.e. empty string) nor any files
+            to upload.
+        AttributeError: `post` has not enough or too many attributes.
+    """
+    required_attributes = {"event_id", "author_id", "text", "files"}
+    if post.keys() != required_attributes:
+        raise AttributeError(f"Arg post must have exactly the "
+                             "attributes {required_attributes}")
+    if str(post["text"]) == "" and list(post["files"]) == []:
```

Comment:
Can you use python truthiness here?

```python
if not post["text"] and not post["files"]:
  raise ...
```

Reply:
Yeah, I was overthinking it. Done https://github.com/knative-portability/large-events/pull/16/commits/9020e0fd725fd4c8fe9bb1d0695f6348e2b3936f